### PR TITLE
Docsite sphinx rm buildsite (and speed up docs build)

### PR DESCRIPTION
**Pull request by alikins**
##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

docs/docsite/

##### ANSIBLE VERSION

```
ansible 2.3.0 (docsite_sphinx_rm_buildsite 65bf5713c0) last updated 2017/01/13 13:22:49 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = [u'/home/adrian/ansible/my-modules', u'/home/adrian/ansible/lib/modules/']

```

##### SUMMARY

Stop using docsite/build-site.py and use a Makefile from sphinx-quickstart  as Makefile.sphinx

### CODE REVIEW

**Comment by alikins** [comment]
Notes: This ends up being the docs in a different dir than before (new: docs/docsite/_build/html/rst  old: docs/docsite/htmlout)

**Comment by alikins** [comment]
speedup:

old (even with CPUS=16 for ex) : hours? weeks? 
new (without using CPUS/-j at all):

```
[fedora-25:docsite (docsite_sphinx_rm_buildsite %)]$ time make webdocs
<..>
real	0m58.702s
user	0m56.756s
sys	0m1.244s
```

**Comment by alikins** [comment]
And with 'CPUS=4 make webdocs' it's down to 0:31.73elapsed

File: docs/docsite/Makefile.sphinx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Put it first so that "make" without argument is like "make help".
+help:
+	@$(SPHINXBUILD) -M help "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+.PHONY: help Makefile.sphinx
+
+# Catch-all target: route all unknown targets to Sphinx using the new
+# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
+%: Makefile.sphinx
```
Comment (abadger) [inline docs/docsite/Makefile.sphinx:19]: % patterns are a GNU Make extension.  So we'll need to change this.   Maybe we just need to spell out all the possibilities?  I can't recall a portable match-anything Makefile rule.

Comment (abadger) [reply docs/docsite/Makefile.sphinx:19]: Ah ha!:  .DEFAULT:

http://stackoverflow.com/questions/33009955/catchall-target-with-bsds-make

Comment (alikins) [reply docs/docsite/Makefile.sphinx:19]: Going to push it and see if it breaks docs builds on bsds.  I don't see any mention of problems in the sphinx builds, but one to find out.

**Pull request merged**